gui.py

```diff
@@ -0,0 +1,173 @@
+# gui.py
+# Uses PyQt to collect camera footage and display information from Microcontroller
+# Uses PyQt to collect keyboard input too
+
+import random
+import sys
+
+from PyQt5.QtCore import Qt, QObject, QThread, pyqtSignal, QTimer, QEvent
+import PyQt5.QtMultimedia as QTM
+import PyQt5.QtMultimediaWidgets as QTMW
+import PyQt5.QtWidgets as QT
+from PyQt5.QtGui import QKeyEvent
+
+from mcu import MCUInterface
+
+sys.path.insert(0, './mcu-lib')
+app = QT.QApplication(sys.argv)
+
+
+class MainWindow(QT.QWidget):
+    def __init__(self, mcuobject: MCUInterface, comms: Communications, exit_program: ExitProgram):
+        super().__init__()
+        self.mcu: MCUInterface = mcuobject
+        self.thruster1speed: int = 0
+        self.thruster2speed: int = 0
+        self.thruster3speed: int = 0
+        self.thruster4speed: int = 0
+        self.servospeed: int = 0
+        
+        self.comms: Communications = comms
+        self.exit_program: ExitProgram = exit_program
+        
+        # (key, message sent to comms)
+        self.keys_pressed = [
+            ("W", "lw"),
+            ("w", "w"),
+            ("A", "la"),
+            ("a", "a"),
+            ("S", "ls"),
+            ("s", "s"),
+            ("D", "ld"),
+            ("d", "d"),
+            ("E", "le"),
+            ("e", "e"),
+            ("Q", "lq"),
+            ("q", "q"),
+            (" ", "spacebar")
+        ]
+        
+        self.keys_released = [
+            ("w", "sw"),
+            ("a", "sa"),
+            ("s", "ss"),
+            ("d", "sd"),
+            ("e", "se"),
+            ("q", "sq")
+        ]
+    
+    def updatetext(self):
+        self.voltage_info.setText(str(self.mcu.latest_voltage))
+        self.x_gyro.setText(str(self.mcu.latest_gyro[0]))
+        self.y_gyro.setText(str(self.mcu.latest_gyro[1]))
+        self.z_gyro.setText(str(self.mcu.latest_gyro[2]))
+        self.x_accel.setText(str(self.mcu.latest_accel[0]))
+        self.y_accel.setText(str(self.mcu.latest_accel[1]))
+        self.z_accel.setText(str(self.mcu.latest_accel[2]))
+        self.temperature.setText(str(self.mcu.latest_temp))
+        self.thruster1.setText(str(self.thruster1speed))
+        self.thruster2.setText(str(self.thruster2speed))
+        self.thruster3.setText(str(self.thruster3speed))
+        self.thruster4.setText(str(self.thruster4speed))
+        self.servo.setText(str(self.servospeed))
+    
+    def setupui(self):
+        TITLE: str = 'MATE'
+        X_POSITION: int = 0
+        Y_POSITION: int = 0
+        LENGTH: int = 1600
+        WIDTH: int = 800
+        
+        self.setWindowTitle(TITLE)
```

Comment:
modularize this chunk, between lines 81 and 147. use private methods (`__method`)